Dockerfile

```diff
@@ -0,0 +1,7 @@
+# Pin to Alpine 3.9
+# For a complete list of hashes, see:
+# https://github.com/docker-library/repo-info/tree/master/repos/alpine/remote
+FROM alpine@sha256:115731bab0862031b44766733890091c17924f9b7781b79997f5f163be262178
+RUN mkdir /usr/agent
+ARG AGENT_DIR
+COPY ${AGENT_DIR} /usr/agent/nodejs
```

Comment:
For comparison the Java APM agent docker container is using "/usr/agent/elastic-apm-handler{,.jar}". With this the Node.js APM agent will be at "/usr/agent/nodejs/node_modules/elastic-apm-node/...", I think (I haven't run this build yet :).

Reply:
I don't know exactly how the k8s attachment will work.  I'm guessing it will mount one of these directories in a target container and add a `NODE_OPTIONS="--require $mountPath"` envvar.

I *think* that would work without the `/node_modules/` component in the path. In other words, I think it might be possible to have the agent Docker image use "/usr/agent/elastic-apm-node/..." for its path, if you want.

Reply:
https://github.com/elastic/apm-agent-nodejs/issues/2622#issuecomment-1131573905 mentions perhaps wanting this Docker image to be also usable for container-based Lambda functions. That might dictate what path we want the APM agent to be installed into.

Reply:
TL;DR; -- the path the agent's copied to in the container is arbitrary and its path doesn't need a particular name.  

Re: k8s attachment

> I don't know exactly how the k8s attachment will work. I'm guessing it will mount one of these directories in a target container and add a NODE_OPTIONS="--require $mountPath" envvar.

That's almost it.  The additional nuance is that the agent's source will be copied out of the Docker container and into a mounted volume at a fixed path.  The command that currently does this copying is here: https://github.com/elastic/apm-mutating-webhook/blob/2c6a4cecf33fb8022eb1f3a365e6705345ff908a/patch.go#L150.  (part of the work of K8s attachment is changing this copy to be able to work with multiple files instead of a single file `jar`) The some `NODE_OPTIONS` and `NODE_PATH` setting will happen in order to load the agent from this spot.

Reply:
Thanks.
We coudl consider *just* setting/updating `NODE_OPTIONS` and not bothering with `NODE_PATH`, right?

Reply:
> We could consider just setting/updating NODE_OPTIONS and not bothering with NODE_PATH, right?

That would be ideal -- but without a `NODE_PATH` set (it doesn't seem like?) manual instrumentation in the user's code would able to find the agent module.  Is that incorrect?

Reply:
> manual instrumentation in the user's code would able to find the agent module. Is that incorrect?

You are correct. I think there are some finer points around that that might be easier to discuss on a call (or better on https://github.com/elastic/apm-agent-nodejs/issues/2622). Basically whether we *want* the user to be able to `require('elastic-apm-node')` and get this externally installed package. My (limited) understanding of the OTel equivalent is that they are not modifying `NODE_PATH`.  If the programmer is adding manual instrumentation then they are likely better served having their own `"elastic-apm-node"` in their package.json dependencies.